File: 1686-stone-game-vi/1686-stone-game-vi.cpp
class Solution {
public:
    int stoneGameVI(vector<int>& alice, vector<int>& bob) {
        int n = alice.size();
        priority_queue<pair<int, int>> q;
        for(int i = 0; i<n; ++i){
            q.push({alice[i]+bob[i], i});
        }
        int sum = 0;
        for(int i = 0; i<n; ++i){
            auto top = q.top();
            q.pop();
            if(i&1) sum -= bob[top.second];
            else sum += alice[top.second];
        }
        if(sum)
        sum /= abs(sum);
        return sum;
    }
};

File: 1686-stone-game-vi/1686-stone-game-vi.java
class Solution {
    public int stoneGameVI(int[] alice, int[] bob) {
        int n = alice.length;
        Integer[] idx = IntStream.range(0, n).boxed()
        .sorted((i, j) -> ((alice[j]+bob[j]) -(alice[i]+bob[i])))
        .toArray(Integer[]::new);

        int sum = 0;
        for(int i = 0; i<n; ++i){
            if((i&1) == 0)
                sum += alice[idx[i]];
                else sum -= bob[idx[i]];
        }
        return Integer.compare(sum, 0);
    }
}